working-groups/meta/rfc-drafts/compiler-team-contributors.md

```diff
@@ -0,0 +1,244 @@
+the like, so that they need not be bothered with pings and messages.
+They will also not have r+ privileges. **Alumni members will however
+still remain members of the GitHub org overall.**
+
+People in alumni status can ask to return to "active" status at any
+time.
+
+People in alumni status are still members of the team at the level
+they previously attained and they may publicly indicate that, though
+they should indicate the time period for which they were active as
+well.
+
+## Automatic alumni status after 6 months of inactivity
+
+If a contributor or a member has been inactive in the compiler for 6
+months, then we will ask them if they would like to go to alumni
+status. If they respond yes or do not respond, they can be placed on
+alumni status.  If they would prefer to remain active, that is also
+fine, but they will get asked again periodically if they continue to
+be inactive.
+
+# Drawbacks
+[drawbacks]: #drawbacks
+
+Why should we *not* do this?
+
+# Rationale and alternatives
+[rationale-and-alternatives]: #rationale-and-alternatives
+
+This RFC represents, effectively, the smallest extension to our structure
+that could possibly work. One could imagine more elaborate structures along a few dimensions.
+
+**More senior levels of membership.** One concern is that the set of
+**members** of the compiler team may grow too large for things like
+FCP (where each person must check their box) to be feasible. This
+could be resolved by moving away from FCP-based decision making (which
+is rarely used in the compiler anyhow), but it may also be worth
+considering another level of membership (e.g., a **senior
+member"). Senior members could be used for FCP-level decisions, which
+would presumably be relatively rare. At present there is a kind of
+implicit amount of "seniority" amongst members, where the opinions of
+people who have been around for longer are obviously given great
+weight, but formalizing this could have value.
+
+**Specialists and organizers.** Right now, we don't draw a distinction
+between people who write code and those who (for example) perform more
+organizational roles (as of the time of this writing, we don't have
+any members who perform more organizational roles exclusively, but
+that is a likely future development). There will definitely be
+contributors who would rather not participate in the more
+organizational aspects of running the team, but would prefer to simply
+write code. As the team gets more and more organized, it seems likely
+that we may want to recognize this distinction, just to avoid things
+like pinging folks with organizational questions when they are not
+interested in that. But we could also address this by growing more
+kinds of groups within the set of members, such that one rarely pings
+the full set of members.
+
+# Prior art
+[prior-art]: #prior-art
+
+It would be good to include some survey of how other open source
+organizations manage themselves here.
```

Comment:
Maybe ask around in GHC circles and whatnot?

cc @goldfirere @simonpj

Reply:
We don't really have a formal process for this. When someone starts contributing a lot, we grant them more privileges. To be honest, @bgamari may have a more informed opinion here than I do. What has provoked the need for formalizing this? One downside of our informal process is that it may be hard for an individual without a connection to an "insider" to get promoted through the ranks... but I don't think that's played out in practice. (Then again, I'm biased, as an insider.)

Reply:
> What has provoked the need for formalizing this?

Mostly a matter of scale I think; but @nikomatsakis probably knows best re. the needs of Rust's compiler team... :)

Reply:
Thanks @goldfirere for the input. Loosely speaking, the desire to formalize is to help make the path to contribution clearer, and also just to make sure all of our expectations are aligned. As the motivation of the RFC states:

> This proposal is part of a larger effort to introduce more structure
> into the compiler team's makeup. This structure should make it easier
> to become a part of the compiler team, by laying out a clear path to
> membership and by offering more official roles.

I've added a few notes based on what you said to the "Prior art" section. =)